lib/improver/nowcasting/pysteps_advection.py

```diff
@@ -72,8 +85,13 @@ def _get_precip_rate(self):
             np.ndarray:
                 2D precipitation rate array in mm h-1
         """
-        self.analysis_cube, = ApplyOrographicEnhancement("subtract").process(
-            self.analysis_cube, self.orogenh)
+        if self.orogenh:
+            self.analysis_cube, = ApplyOrographicEnhancement(
+                "subtract").process(self.analysis_cube, self.orogenh)
+        elif "precipitation_rate" in self.analysis_cube.name():
+            msg = ("For precipitation fields, orographic enhancement "
+                   "cube must be supplied.")
+            raise ValueError(msg)
```

Comment:
I'm going to say we don't need this here, at least not without some other changes.  The PySteps extrapolation code is designed to work with precipitation rates in mm/h, and this wrapper plugin has been written with that assumption embedded - this function for example is `_get_precip_rate`, and gets them in mm/h for pysteps' consumption.  I can't think of any situation in which we would not need an orographic enhancement cube input here.

Reply:
I take your point about PySteps being designed for precipitation rate, but I can think of fields I would like to advect to produce a nowcast that are not precipitation but are closely linked, e.g. cloud, wind gust. I think it is important to use the same advection method for these so I expect that option to be available. My understanding is that the units are only important to the Optical Flow algorithm, the advection shouldn't care.

Reply:
I couldn't find anything in the PySteps extrapolate method that said what the units of the input array should be, but there isn't for any of the other fields either. As the input field is named "precip", I have removed this option for now.